### PR TITLE
add pii_verify_engine

### PR DESCRIPTION
**Pull request by zhangabner**
add pii_verify_engine and test.
it could be usefule to verify the pii before redacting them.  It is similar to redacting, but add empty box on word and add type of pii above it by using ax.annotation.

I use Automatically format code and check for code styling in the guideline, my codes is the bottom part of changes which passed the test, but it also change some other part of code in top changes

### CODE REVIEW

**Comment by omri374** [comment]
Thanks for your PR @zhangabner! We will go over it soon.

**Comment by omri374** [comment]
Hi @zhangabner, there are merge conflicts. Could you please fix them so that we could properly review this?

**Comment by zhangabner** [comment]
pre-commit install  changed  https://github.com/microsoft/presidio/blob/main/e2e-tests/tests/test_anonymizer.py
I restored the oringal file

**Comment by zhangabner** [comment]
pre-commit install  changed e2e-tests/tests/test_anonymizer.py  I restored the original file.

File: presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from PIL import Image, ImageChops
+from presidio_image_redactor.image_analyzer_engine import ImageAnalyzerEngine
+import matplotlib
+import io
+from matplotlib import pyplot as plt
```
Comment (omri374) [inline presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: Is matplotlib a necessity here? We'd rather not add additional dependencies unless mandatory. Can we use PIL for this logic?

Comment (zhangabner) [reply presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: Hi, I use ax.annotate to draw entity_type above the empty box of the words, then transfer fig to image.
      fig, ax = plt.subplots()
      rect = matplotlib.patches.Rectangle(
                    (x0, y0), x1 - x0, y1 - y0, edgecolor="b", facecolor="none"
                )
      ax.annotate(
                    entity_type,
                    xy=(x0 - 3, y0 - 3),
                    xycoords="data",
                    bbox=dict(boxstyle="round4,pad=.5", fc="0.9"),
                )
    
       im_from_fig = fig2img(fig)

Comment (omri374) [reply presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: @zhangabner in this case could you please add matplotlib to setup.py and Pipfile?

Comment (zhangabner) [reply presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: @omri374 sure, I added matplotlib to setup.py and Pipfile.

Comment (navalev) [reply presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: @zhangabner I see there's some merge conflicts. Could you please resolve them?

Comment (zhangabner) [reply presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:5]: @navalev , resolved.

**Comment by omri374** [comment]
Hi @zhangabner, there are some linting issues. Could you please take a look?

**Comment by zhangabner** [comment]
hi, @omri374 , I also noticed the linting issue.
I followed Automatically format code and check for code styling,  it automatically changed some code like https://github.com/microsoft/presidio/blob/main/e2e-tests/tests/test_anonymizer.py
is it caused by the version or the config of pre-commit?

**Comment by omri374** [comment]
Here are the problematic lines:
```
./presidio_image_redactor/__init__.py:5:1: F401 '.tesseract_ocr.TesseractOCR' imported but unused
./presidio_image_redactor/__init__.py:14:88: W292 no newline at end of file
```

the CI runs `pipenv run flake8`

**Comment by zhangabner** [comment]
./presidio_image_redactor/__init__.py:5:1: F401 '.tesseract_ocr.TesseractOCR' imported but unused
this line is in the original file,  do you want me to delete it?

**Comment by omri374** [comment]
You removed it from the `__all__` so it's complaining

**Comment by zhangabner** [comment]
updated

**Comment by omri374** [comment]
@zhangabner there are still linting issues. Could you please run `pipenv run flake8` to make sure everything is resolved? More info can be found on the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md#how-to-test).

**Comment by omri374** [comment]
hi @zhangabner. Do you have visibility to the Azure Pipelines on this PR? there are some failing tests.

**Comment by zhangabner** [comment]
Hi, @omri374  yes, I could see the the Azure pipeline.
test_given_image_with_tuple_color_filling_then_we_get_redact_image
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=5003): Max retries exceeded with url: /redact (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc064200b20>: Failed to establish a new connection: [Errno 111] Connection refused'))

**Comment by zhangabner** [comment]
and many files are changed because I run pre-commit install
image

**Comment by zhangabner** [comment]
the only two errors is in image redact test, which are in the original files and I didn't change it.

**Comment by omri374** [comment]
@zhangabner we'll take a look. thanks

**Comment by SharonHart** [comment]
@zhangabner 
The image redactor e2e tests are failing since the container fails to start with:
`Traceback (most recent call last):
  File "app.py", line 8, in <module>
    from presidio_image_redactor import ImageRedactorEngine
  File "/usr/bin/presidio-image-redactor/presidio_image_redactor/__init__.py", line 8, in <module>
    from .image_pii_verify_engine import ImagePiiVerifyEngine
  File "/usr/bin/presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py", line 3, in <module>
    import matplotlib
ModuleNotFoundError: No module named 'matplotlib'
`
matplotlib is in the pipfile, but is missing from the lock file, which is the effective file for installing dependencies. 
Our documentation is still lacking proper instructions on how to lock pipenv, but essentially you should run:
`pipenv lock --python 3.6` once satisfied with the pipfile, to generate a pipenv.lock file.
Changes involving changing dependencies are complex, let us know if you need any help with resolving this, or if you want us to add the dependency desperately. We would really be happy to see this change added to the image redactor 😄

**Comment by zhangabner** [comment]
@SharonHart Thanks for the reply. I am new with lock, pls help to add it to a separate PR. thanks.

**Comment by zhangabner** [comment]
It seems, matplotlib caused the version conflict in numpy. @SharonHart 
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
ImportError while loading conftest '/home/vsts/work/1/s/presidio-image-redactor/tests/conftest.py'.
tests/conftest.py:4: in <module>
    from presidio_image_redactor import ImageAnalyzerEngine
presidio_image_redactor/__init__.py:8: in <module>
    from .image_pii_verify_engine import ImagePiiVerifyEngine
presidio_image_redactor/image_pii_verify_engine.py:3: in <module>
    import matplotlib
/home/vsts/.local/share/virtualenvs/presidio-image-redactor-UNwSIaum/lib/python3.9/site-packages/matplotlib/__init__.py:207: in <module>
    _check_versions()
/home/vsts/.local/share/virtualenvs/presidio-image-redactor-UNwSIaum/lib/python3.9/site-packages/matplotlib/__init__.py:192: in _check_versions
    from . import ft2font
E   ImportError: numpy.core.multiarray failed to import

**Comment by navalev** [comment]
@zhangabner I'll update your branch with the latest fixes and re-run the build pipeline

**Review by navalev — APPROVED** [review]

File: presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class ImagePiiVerifyEngine:
+    """ImagePiiVerifyEngine class only supporting Pii verification currently."""
+
+    def __init__(self):
+        self.analyzer_engine = ImageAnalyzerEngine()
+
+    def verify(self, image: Image) -> Image:
+        """Redact method to redact the given image.
```
Comment (omri374) [inline presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:25]:
```suggestion
        """Annotate image with the detect PII entity.
```
Or something similar :)

File: presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        Please notice, this method duplicates the image, creates a new instance and
+        manipulate it.
+        :param image: PIL Image to be processed
+        :return: the redacted image
```
Comment (omri374) [inline presidio-image-redactor/presidio_image_redactor/image_pii_verify_engine.py:30]:
```suggestion
        :return: the annotated image
```

**Comment by zhangabner** [comment]
Thanks for the advice. @omri374 . I updated the code based on your suggestions.

**Review by omri374 — APPROVED** [review]

**Comment by omri374** [comment]
Thanks @zhangabner! This is super cool. @navalev could you please re-review?

**Review by navalev — APPROVED** [review]

**Pull request merged**